### DISCUSSION TITLE
Switch from java.util.Date to java.time.Instant?

### DISCUSSION THREAD

**Discussion by icecreamhead** [jwtk/jjwt#904] [Q&A] [answered]
Is there any plan to switch from the deprecated java.util.Date to the more modern java.time.Instant?

**Comment by lhazlewood** [comment, marked as answer]
Most definitely, but as that forces a >= JDK 8 dependency, we were waiting for 1.0 compliance with the RFCs before doing so.  The only thing remaining is for JJWT to be able to parse full JSON JWS and JWE documents (i.e. non-compact JWT).  Once that's done, we were going to release a final 0.x version (for JDK 7) and 1.0+ for JDK 8 and later.  Maybe even JDK 11, 15, or 17 as the baseline - we haven't decided yet.

**Comment by m-aqsam** [comment]
As of my last knowledge update in January 2022, Java has already deprecated java.util.Date and related classes in favor of the more modern and comprehensive java.time API introduced in Java 8. Therefore, in the short term, it is recommended to switch from java.util.Date to java.time.Instant and other java.time classes for better date and time handling.

**Reply by lhazlewood** [reply]
You're definitely correct, but, again, we can only do this after 1.0 compliance with the RFCs (which is only full JSON documents, not compact tokens).  After that's done, we 100% will migrate to `java.time` classes.
